janitor/functions.py

```diff
@@ -97,6 +97,7 @@ def clean_names(
         Only letters, numbers and underscores are preserved.
     :returns: A pandas DataFrame.
     """
+    mylist = list(df.columns)
```

Comment:
I'd suggest doing the following:

1. Firstly, make preservation of names optional, by putting it in the `clean_names` function signature.
2. Check to see if `preserve_names` is set too `True`. If yes, then store it.

Reply:
As suggested, I have added a parameter called original_columns. The default is True because it does not consume any resources. The user can disable it.